algorithms/aqc/solving_qlsp/solving_qlsp_with_aqc.ipynb

```diff
@@ -0,0 +1,867 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #45.        HO_HAMILTONIAN = matrix_to_hamiltonian(H0)
</pre><p>I think that it is possible to use the prepare_state(b) to directly encode the exp(iH0*t) by using the phase operator, like the grover-diffuser, and the projector_controlled_phase operation in the qsvt.</p><p>You do something like:</p><p>A*B_dag*phase(t)*B*A_dag </p><p>where A = inplace_prepare_state(b, qvar)</p><p>B = aux = (qvar == 0) </p><p>phase(t) = phase(aux, t)</p><p>like:</p>

<pre class="ql-syntax" spellcheck="false">within_apply(lambda: [inverse(inplace_prepare_state(b, 0, qvar),
                      assign(qvar==0, aux)],
             lambda: phase(aux * t, 1))
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/688/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Reply:
<p>Cool!! @orsa-classiq</p>